src/backend/parser/parse_coerce.c

```diff
@@ -336,6 +336,7 @@ coerce_type(ParseState *pstate, Node *node,
 												  ccontext, cformat, location);
 			if (result)
 			{
+				cancel_parser_errposition_callback(&pcbstate);
```

Comment:
In some cases we cancel the error callback inside the hook. [link](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_4_X_DEV/contrib/babelfishpg_tsql/src/pltsql_coerce.c#L1152).

Reply:
I assume this call can be removed from the hook, filed [#2695](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2695) to do so.

Reply:
Please also add @2jungkook to reviewers, I do not fully understand why `cancel_parser_errposition_callback` wasn't called exactly before returning the result.

Reply:
+1

Reply:
`cancel_parser_errposition_callback` is not called for binary/varbinary types (https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_4_X_DEV/contrib/babelfishpg_tsql/src/pltsql_coerce.c#L1185-L1214), hence this change looks good to me. Thank @staticlibs for fixing this!